torch/csrc/jit/OVERVIEW.md

```diff
@@ -886,267 +884,244 @@ def LSTMCellS(x, hx, cx, w_ih, w_hh, b_ih, b_hh):
+  %27 : Tensor = prim::profile[profiled_type=Double(64:2048, 2048:1, requires_grad=1, device=cuda:0)](%gates.1)
+  %28 : Tensor, %29 : Tensor, %30 : Tensor, %31 : Tensor = prim::ConstantChunk[chunks=4, dim=1](%27)
+  %32 : Tensor = prim::profile[profiled_type=Double(64:2048, 512:1, requires_grad=1, device=cuda:0)](%28)
+  %ingate.3 : Tensor = aten::sigmoid(%32) # test_jit_fuser_te.py:84:21
+  %34 : Tensor = prim::profile[profiled_type=Double(64:2048, 512:1, requires_grad=1, device=cuda:0)](%29)
+  %forgetgate.3 : Tensor = aten::sigmoid(%34) # test_jit_fuser_te.py:85:25
+  %36 : Tensor = prim::profile[profiled_type=Double(64:2048, 512:1, requires_grad=1, device=cuda:0)](%30)
+  %cellgate.3 : Tensor = aten::tanh(%36) # test_jit_fuser_te.py:86:23
+  %38 : Tensor = prim::profile[profiled_type=Double(64:2048, 512:1, requires_grad=1, device=cuda:0)](%31)
+  %outgate.3 : Tensor = aten::sigmoid(%38) # test_jit_fuser_te.py:87:22
+  %40 : Tensor = prim::profile[profiled_type=Double(64:512, 512:1, requires_grad=1, device=cuda:0)](%forgetgate.3)
+  %41 : Tensor = prim::profile[profiled_type=Double(64:512, 512:1, requires_grad=1, device=cuda:0)](%cx.1)
+  %42 : Tensor = aten::mul(%40, %41) # test_jit_fuser_te.py:88:18
+  %43 : Tensor = prim::profile[profiled_type=Double(64:512, 512:1, requires_grad=1, device=cuda:0)](%ingate.3)
+  %44 : Tensor = prim::profile[profiled_type=Double(64:512, 512:1, requires_grad=1, device=cuda:0)](%cellgate.3)
+  %45 : Tensor = aten::mul(%43, %44) # test_jit_fuser_te.py:88:38
+  %46 : Tensor = prim::profile[profiled_type=Double(64:512, 512:1, requires_grad=1, device=cuda:0)](%42)
+  %47 : Tensor = prim::profile[profiled_type=Double(64:512, 512:1, requires_grad=1, device=cuda:0)](%45)
+  %cy.1 : Tensor = aten::add(%46, %47, %7) # test_jit_fuser_te.py:88:18
+  %49 : Tensor = prim::profile[profiled_type=Double(64:512, 512:1, requires_grad=1, device=cuda:0)](%cy.1)
+  %50 : Tensor = aten::tanh(%49) # test_jit_fuser_te.py:89:27
+  %51 : Tensor = prim::profile[profiled_type=Double(64:512, 512:1, requires_grad=1, device=cuda:0)](%outgate.3)
+  %52 : Tensor = prim::profile[profiled_type=Double(64:512, 512:1, requires_grad=1, device=cuda:0)](%50)
+  %hy.1 : Tensor = aten::mul(%51, %52) # test_jit_fuser_te.py:89:17
+  %54 : Tensor = prim::profile[profiled_type=Double(64:512, 512:1, requires_grad=1, device=cuda:0)](%hy.1)
+  %55 : Tensor = prim::profile[profiled_type=Double(64:512, 512:1, requires_grad=1, device=cuda:0)](%cy.1)
+  %56 : (Tensor, Tensor) = prim::TupleConstruct(%54, %55)
+```
+
+### Optimization ###
+
+The graph executor performs a number of optimizations.
+The optimizations the graph executor performs can be broadly classified into:
+* optimizations that require profiling information and the ones that do not
+* optimizations that preserve graph's differentiability and post-derivative optimizations
+
+.|Derivative-preserving|Post-derivative
+--|--|--
+**Profiling**|Peephole* </br> | Tensor Expressions Fuser </br> CUDA Fuser 
```

Comment:
I'm kind of trying to imagine what all this formatting will look like. I am reading github view..

Reply:
You can click "Display the rich diff" and it'll render the formatting:
![image](https://user-images.githubusercontent.com/4588111/96942695-b5055500-148a-11eb-8cca-bc5c262cefed.png)